### PR TITLE
Support libpeas based plugin system

### PR DESCRIPTION
**Pull request by Rafostar**
Add a `libpeas` based plugin system. In order to not be confused with GStreamer plugins, I will be using term `enhancers` as these are meant to be callable from already present functionalities in Clapper library, enhancing them (possibly using other non-C libraries). These are meant mostly for either not actual playback and/or out of GStreamer scope functionalities.

Currently implemented is `CLAPPER_EXTRACTABLE`, an interface for creating enhancers that extract data from different sources before letting it be played via GStreamer. Note that absolutely no actual extraction code is shipped as part of the app. For this reason loadable `enhancers` system was invented.

For now, this is meant to replace `gtuber` with something (hopefully) easier to maintain by using extractors from different libraries (that others will maintain instead of myself :laughing:).

---

**Issues:**
Needs `CLAPPER_USE_PLAYBIN3=1` to play smoothly in HQ - not fixable here, we should change to playbin3 by default for next Clapper release anyway while other apps can already enforce it with env, so not a problem.

**Testing:**
Both local build and Flatpak package from this PR CI (checks tab) work fine. Windows is currently untested.

### CODE REVIEW

**Comment by Rafostar** [comment]
Currently, applications can quickly check an `enhancer` availability with:
```c
gboolean
clapper_enhancer_check (
  GType iface_type,
  const gchar* scheme,
  const gchar* host,
  const gchar** name
)
```
AND check whether Clapper was even compiled with ability to load enhancers (requires `libpeas`) by using `CLAPPER_WITH_ENHANCERS_LOADER` or alternatively get this from `pkgconfig`.

Example:
```c
gboolean supported = clapper_enhancer_check (CLAPPER_TYPE_EXTRACTABLE, "https", "example.com", NULL);
```

As requested in: https://github.com/GeopJr/Tuba/pull/1171

ping @GeopJr

**Comment by GeopJr** [comment]
Sounds good to me, thanks! And I'm interested in maintaining a few extractors probably :)

File: src/lib/clapper/clapper-harvest.c [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+ * NOTE: We cannot simply expose GstMiniObjects for
+ * implementations to assemble TagList/Toc themselves, see:
+ * https://gitlab.freedesktop.org/gstreamer/gstreamer/-/issues/2867
+ */
+
+#include "clapper-harvest-private.h"
+
+#define REQUEST_HEADERS_STRUCTURE_NAME "request-headers"
+#define UA_FIELD_NAME "User-Agent"
+#define DEFAULT_UA "Mozilla/5.0 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)"
```
Comment (GeopJr) [inline src/lib/clapper/clapper-harvest.c:38]: Should the default UA be googlebot-like, since this is not a crawler?

**Comment by Rafostar** [comment]
Created a separate repo for enhancer type of plugins [here](https://github.com/Rafostar/clapper-enhancers). Also added there an `yt-dlp` enhancer implementing `ClapperExtractable` interface from this PR.

I will leter update this PR to fix CI with missing dependencies and replace `gtuber` with this new stuff, so it will be easier to test.

File: src/lib/clapper/clapper-harvest.c (hunk 1 above)
Comment (Rafostar) [reply src/lib/clapper/clapper-harvest.c]: Good point. Changed to Tor browser default. This is meant only as fallback if plugin does not set any.

**Comment by Rafostar** [comment]
> Sounds good to me, thanks! And I'm interested in maintaining a few extractors probably :)

Thanks. Hopefully now most of the hard stuff will be handled by `Clapper Enhancer yt-dlp` and `yt-dlp` guys will maintain actual web extraction for us :smile:.

Nonetheless, you are (and everyone else) always welcome to contribute (maybe even with new extractors). Now they can be written in `C`, `Vala`, `Python`, `GJS` and maybe even `Rust`. I will be moving the ones that are preferred to be custom written over `yt-dlp` from `gtuber` repo.

**Comment by Rafostar** [comment]
The CI now passed, Flatpak was built and... it works! :tada:
I will be updating this PR first comment description with any issues/concerns/todo's I find while testing.

File: src/lib/clapper/clapper-harvest.c (hunk 1 above)
Comment (Rafostar) [reply src/lib/clapper/clapper-harvest.c]: Ultimately, I removed this part completely. GStreamer will set us a program-like header so requests will work during actual request construction and this part should not enforce headers when not doing a web extraction.

**Comment by Rafostar** [comment]
OK. So what was tested overall now:

- [x]  loading/writing python plugins
- [x] loading/writing C (compiled) plugins
- [x] custom URI schemes
- [x] passing tags, TOC and headers
- [x] DASH streaming
- [x] HLS streaming
- [x] direct video URI playback

In overall, everything "here" seems to work fine. What remains is to add/restore more extractors, but these can be even gradually added later (even after next Clapper version), since its another repo and `gtuber` can still work as a fallback when installed for the not yet ported things.

**Pull request merged**